tests/Doctrine/Tests/ORM/Functional/Ticket/GH7692Test.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH7692Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH7692Project::class,
+            GH7692Contact::class,
+        ]);
+    }
+
+    public function testWrongForeignKeysInDatabaseAreHandledByDoctrine(): void
+    {
+        // Create a row that references missing rows
+        $this->_em->getConnection()->insert('project', [
+            // This composite foreign key doesn't exist
```

Comment:
This seems to be an impossible scenario from a relational DB perspective. In practice, you'd get a proxy lazy-loading exception when not operating with `EAGER` loading

Reply:
Understood, I was under the false impression that it worked without eager. It actually throws an exception like you say.

However the exception makes sense: it's a `EntityNotFoundException`.

I have updated the PR:

- added an additional test for a scenario that currently works (throws `EntityNotFoundException`)
- updated the existing test to assert that a `EntityNotFoundException` is thrown

The test still fails as the current `spl_object_hash` warning isn't what we want.

Reply:
I have given a try to a fix (to have a proper exception).